pkg/controllers/machine/garbagecollect.go

```diff
@@ -0,0 +1,48 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machine
+
+import (
+	"context"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+)
+
+type GarbageCollect struct {
+	kubeClient    client.Client
+	cloudProvider cloudprovider.CloudProvider
+	lastChecked   *cache.Cache
+}
+
+func (e *GarbageCollect) Reconcile(ctx context.Context, machine *v1alpha5.Machine) (reconcile.Result, error) {
+	// If there is no node representation for the machine, then check if there is a representation at the cloudprovider
+	if _, err := nodeForMachine(ctx, e.kubeClient, machine); err == nil || !IsNodeNotFoundError(err) {
```

Comment:
I was thinking that we could just solely rely on the node existence but there is a case where someone doesn't have ttlAfterNotRegistered set and we would want to kill the machine if the instance was gone.